src/check.h.in

```diff
@@ -669,6 +671,129 @@ CK_DLL_EXP void CK_EXPORT _ck_assert_failed(const char *file, int line,
+#define _ck_assert_floating(X, OP, Y) do { \
+  long double _ck_x = (X); \
+  long double _ck_y = (Y); \
+  ck_assert_msg(_ck_x OP _ck_y, "Assertion '%s' failed: %s == %Lg, %s == %Lg", #X" "#OP" "#Y, #X, _ck_x, #Y, _ck_y); \
+} while (0)
+
+/**
+ * Check two floating point numbers to determine if X==Y
+ *
+ * If not X==Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_eq(X, Y) _ck_assert_floating(X, ==, Y)
+/**
+ * Check two floating point numbers to determine if X!=Y
+ *
+ * If not X!=Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_ne(X, Y) _ck_assert_floating(X, !=, Y)
+/**
+ * Check two floating point numbers to determine if X<Y
+ *
+ * If not X<Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_lt(X, Y) _ck_assert_floating(X, <, Y)
+/**
+ * Check two floating point numbers to determine if X<=Y
+ *
+ * If not X<=Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_le(X, Y) _ck_assert_floating(X, <=, Y)
+/**
+ * Check two floating point numbers to determine if X>Y
+ *
+ * If not X>Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_gt(X, Y) _ck_assert_floating(X, >, Y)
+/**
+ * Check two floating point numbers to determine if X>=Y
+ *
+ * If not X>=Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_ge(X, Y) _ck_assert_floating(X, >=, Y)
+
+/**
+ * Check two floating point numbers to determine if X≈Y with specified precision
+ *
+ * If not X≈Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ * @param PREC positive number of decimal places to be compared
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_eq_prec(X, Y, PREC) \
+do { \
+  long double _ck_x = (X); \
+  long double _ck_y = (Y); \
+  int _ck_p = (PREC); \
+  int _prec = -_ck_p; \
+  ck_assert_msg(_prec <= 0, \
```

Comment:
If this assertion is to help protect the usage of this macro consider changing its check to something that relates directly to the user's input. That is,
```
ck_assert_msg((PREC) > 0, ...)
```
This will help a user make sense of the failure.